src/ua_parser/user_agent_parser.py

```diff
@@ -37,6 +37,12 @@ def Parse(
 
             if self.v1_replacement:
                 v1 = self.v1_replacement
+            # FIXME: should be self.user_agent_re.groups (or
+            #        match.re.groups) which actually counts the number
+            #        of capturing groups in the pattern, lastindex
+            #        returns what the match for the last capture
+            #        character is, which is 1 for e.g. (()()) instead
+            #        of 3
```

Comment:
Can probably remove all of these, the comments & formatting fixes were made when the legacy parsers got moved out, before moving them back in without backing out the comments.